store/sqlstore/thread_store.go

```diff
@@ -133,17 +137,57 @@ func (s *SqlThreadStore) GetThreadsForUser(userId string, opts model.GetUserThre
 	if opts.PageSize == 0 {
 		pageSize = opts.PageSize
 	}
+
+	mentionsQuery, mentionsQueryArgs, _ := s.getQueryBuilder().
+		Select("SUM(ThreadMemberships.UnreadMentions)").
+		From("ThreadMemberships").
+		Join("Posts ON Posts.RootId = ThreadMemberships.PostId").
+		Join("Channels ON Posts.ChannelId = Channels.Id").
+		Where(fetchConditions).
+		Where("Posts.UpdateAt >= ThreadMemberships.LastViewed").
+		GroupBy("ThreadMemberships.UserId, ThreadMemberships.Following, Posts.DeleteAt").ToSql()
+	repliesQuery, repliesQueryArgs, _ := s.getQueryBuilder().
+		Select("COUNT(DISTINCT Posts.Id)").
+		From("Posts").
+		Join("ThreadMemberships ON Posts.RootId = ThreadMemberships.PostId").
+		Join("Channels ON Posts.ChannelId = Channels.Id").
+		Where(fetchConditions).
+		Where("Posts.UpdateAt >= ThreadMemberships.LastViewed").
+		GroupBy("ThreadMemberships.UserId, ThreadMemberships.Following, Posts.DeleteAt").ToSql()
+	threadsQuery, threadsQueryArgs, _ := s.getQueryBuilder().
+		Select("COUNT(DISTINCT ThreadMemberships.PostId)").
+		Join("Posts ON Posts.RootId = ThreadMemberships.PostId").
```

Comment:
You can optain the ChannelId to join with the channels table using the Thread table instead of the Posts table, that is going to be less crowded.